### PR TITLE
[RSDK-7081] Simplify control flow in receiveAndWriteSerial

### PR DESCRIPTION
**Pull request by penguinland**
I strongly urge you to review this commit-by-commit! and for the commits that mention flattening the code, review while ignoring whitespace changes.

No changes to behavior are intended: this is just some cleanup of parts I had misunderstood and had to ask Susmita about.

Tried on piworld: works like normal.

### CODE REVIEW

File: components/movementsensor/gpsrtkserial/gpsrtkserial.go [hunk 1]
```diff
@@ -416,13 +416,15 @@ func (g *rtkSerial) receiveAndWriteSerial() {
...
+			continue // No errors: we're still connected.
 		}
 
+		// If we get here, we encountered an error. Try reconnecting to the mount point.
 		isConnectedToNtrip = false
 
 		if msg != nil {
-			continue // We're still connected!
+			// Why would we have a non-nil message with a non-nil error!? Give up entirely.
+			return
```
Comment (penguinland) [inline components/movementsensor/gpsrtkserial/gpsrtkserial.go:422]: The insight for why this didn't change the behavior is that we set `isConnectedToNtrip` to false before continuing, but the loop we're in is `for isConnectedToNtrip`, so we'd stop the loop and return from the function entirely.

**Comment by penguinland** [comment]
Insight: the code in the serial and I2C implementations were originally exact copies of each other. I should do these same changes over there, too. So, the PR just doubled in size!

I still strongly recommend reviewing commit-by-commit.

File: components/movementsensor/gpsrtkserial/gpsrtkserial.go [hunk 2]
```diff
@@ -403,56 +402,50 @@ func (g *rtkSerial) receiveAndWriteSerial() {
...
+	for !g.isClosed {
 		select {
 		case <-g.cancelCtx.Done():
 			return
 		default:
 		}
 
+		// Despite this looking like we're getting the next message from the scanner, what we're
+		// really doing is just checking if the scanner is still connected to the mount point. If
+		// it's disconnected, we'll get an error, which is our signal to reconnect.
```
Comment (susmitaSanyal) [inline components/movementsensor/gpsrtkserial/gpsrtkserial.go:414]: The comment is a bit misleading. According to the [go rtcm package](https://pkg.go.dev/github.com/go-gnss/rtcm/rtcm3#Scanner) Scanner is used for parsing RTCM messages from a reader, which is what we are doing in scanner.NextMessage()

Comment (susmitaSanyal) [reply components/movementsensor/gpsrtkserial/gpsrtkserial.go:414]: If you really want to add a comment here, I would recommend that you say something similar to what the description of NextMessage() is saying

**Review by susmitaSanyal — CHANGES_REQUESTED** [review]
Comments about comments. It's for both I2C and serial

File: components/movementsensor/gpsrtkserial/gpsrtkserial.go (hunk 2 above)
Comment (penguinland) [reply components/movementsensor/gpsrtkserial/gpsrtkserial.go]: How does this new version of the comment sound?

Also, thanks for pointing out that we can never have a non-nil message and non-nil error at the same time: that let me delete a paragraph of dead code, which let me remove the `msg` variable entirely, which helps clarify why there's a comment here.

**Review by susmitaSanyal — APPROVED** [review]
LGTM! 
Thanks for simplifying the workflow

**Pull request merged**